app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -1,14 +1,11 @@
 module Foreman::Model
   class EC2 < ComputeResource
-    has_one :key_pair, :foreign_key => :compute_resource_id, :dependent => :destroy
-
+    include KeyPairComputeResource
+    prepend KeyPairCapabilities
```

Comment:
does it work if the prepend is within KeyPairComputeResource mixin too? (if it does, maybe move it inside the mixin)